kafka_utils/kafka_check/commands/under_replicated.py

```diff
@@ -0,0 +1,135 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+
+import argparse
+import json
+
+import requests
+
+from kafka_utils.kafka_check import status_code
+from kafka_utils.kafka_check.commands.command import get_broker_id
+from kafka_utils.kafka_check.commands.command import KafkaCheckCmd
+
+
+DEFAULT_JOLOKIA_PORT = 8778
+
+
+class UnderReplicatedCmd(KafkaCheckCmd):
+
+    def build_subparser(self, subparsers):
+        subparser = subparsers.add_parser(
+            'under_replicated',
+            description='Check under replicated partitions for all brokers in cluster.',
+            help='This command will sum all under replicated partitions for each broker if any.'
+                 'It will query jolokia port for receive this data.',
+        )
+        subparser.add_argument(
+            '--first-broker-only',
+            action='store_true',
+            help='If this parameter is specified, it will do nothing and succeed on '
+                 'not first brokers from Kafka cluster. Set --broker-id to -1 to read broker-id '
+                 'from --data-path. Default: %(default)s',
+        )
+        subparser.add_argument(
+            '--jolokia-port',
+            type=convert_to_port,
+            default=DEFAULT_JOLOKIA_PORT,
+            help='Port to access to jolokia on each broker. Default: %(default)s',
+        )
+
+        return subparser
+
+    def run_command(self):
+        """Under_replicated command, checks number of under replicated partitions for
+        all brokers in the Kafka cluster."""
+        broker_list = self.zk.get_brokers()
+
+        if self.args.first_broker_only:
+            if self.args.broker_id is None:
+                return status_code.WARNING, 'Broker id is not specified'
+
+            if not check_run_on_first_broker(broker_list, self.args.broker_id, self.args.data_path):
```

Comment:
We want run this check only on one kafka broker. First idea was run it on controller node.
But during restart controller can be changed many times. So, we decided to run it only on the first broker from broker list from Zookeeper.